File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/basics/named_arguments/java/NamedArgumentsExample.java
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.basics.named_arguments.java;

import pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.basics.readability.java.User;

import java.util.List;

public class NamedArgumentsExample {

    public User create() {
        // Java has no named arguments.
        // The meaning depends only on parameter order.
        return new User(
                "John",
                "Doe",
                30,
                List.of("USER", "ADMIN")
        );
    }
}

File: backend-systems-lab-stage-1/src/main/java/pl/jakubtworek/backend_systems_lab_stage_1/block_a/kotlin/basics/named_arguments/kotlin/NamedArgumentsExample.kt
package pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.basics.named_arguments.kotlin

import pl.jakubtworek.backend_systems_lab_stage_1.block_a.kotlin.basics.readability.kotlin.User

class NamedArgumentsExample {

    fun create(): User {
        // Named arguments make method calls easier to read,
        // especially when there are many parameters of the same type.
        return User(
            firstName = "John",
            lastName = "Doe",
            age = 30,
            roles = listOf("USER", "ADMIN")
        )
    }
}